app/scripts/lib/forwardRequestToSnap.ts

```diff
@@ -41,16 +45,22 @@ export async function forwardRequestToSnap(
     throw new InternalError(`No origin specified for method ${method}`);
   }
 
-  const response = (await handleRequest({
-    snapId,
-    origin,
-    handler: HandlerType.OnRpcRequest,
-    request: {
-      jsonrpc: '2.0',
-      method,
-      params,
-    },
-  })) as Json;
+  try {
+    onBeforeRequest?.();
 
-  return response;
+    const response = (await handleRequest({
+      snapId,
+      origin,
+      handler: HandlerType.OnRpcRequest,
+      request: {
+        jsonrpc: '2.0',
+        method,
+        params,
+      },
+    })) as Json;
+
+    return response;
+  } finally {
+    onAfterRequest?.();
```

Comment:
I think if `onAfterRequest()` throws in the finally block, it will completely mask any error from `onBeforeRequest()` or `handleRequest()`, making debugging not easy.